src/qibo/hamiltonians/terms.py

```diff
@@ -199,6 +190,66 @@ def __init__(self, coefficient, factors=1, backend: Optional[Backend] = None):
 
         self.target_qubits = tuple(sorted(self.matrix_map.keys()))
 
+    @staticmethod
+    def _compute_power(factor):
+        """Break down a factor into a base term and its power, e.g. `X**n` returns `X` and `n`"""
```

Comment:
```suggestion
        """Break down a factor into a base term and its power, e.g. `X**n` returns `X` and `n`. In case the base term is a Pauli also simplifies the power calculation, i.e. return the identity for even `n` and the Pauli itsel for odd `n`."""
```
just a small addition